docs/guides/core-internals.mdx

```diff
@@ -0,0 +1,215 @@
+---
+title: Core Internals
+description: A simplified version of the core implementation
+nav: 3.10
+---
+
+This guide is beneficial for those who want to understand the core implementation of the jotai. It is not meant to be a complete guide of the 
+core implementation but rather a simplified version. It is inspired by the collection of tweets by Daishi Kato([@dai_shi](https://twitter.com/dai_shi)).
```

Comment:
Maybe we put links to the collection of tweets for reference?

Reply:
Hi @dai-shi, I have added ref tweet below the implementation. Let me know if this works.

Do we need to add or update any comments in the second implementation ?

Reply:
Thanks. It works.

It looks good. If you/we have ideas to improve it, feel free to raise suggestions.

Reply:
Great! May be we can add a high level architecture overview of its working. Something like redux [does](https://redux.js.org/assets/images/ReduxDataFlowDiagram-49fa8c3968371d9ef6f2a1486bd40a26.gif).

Reply:
Something like this? https://twitter.com/dai_shi/status/1380885195549597696

Reply:
Yes, with animating atom's read write operations and how it re-renders all the subscribed components.

Reply:
yeah, that would be nice.